glusterd2/commands/peers/editpeer.go

```diff
@@ -0,0 +1,144 @@
+package peercommands
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	"github.com/gluster/glusterd2/glusterd2/peer"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/pkg/api"
+
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+)
+
+func editPeer(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	var req api.PeerEditReq
+	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, err.Error())
+		return
+	}
+	for k := range req.MetaData {
+		if strings.HasPrefix(k, "_") {
+			logger.WithField("metadata-key", req.MetaData).Error("Key names starting with '_' are restricted in Metadata field")
+			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "Key names starting with '_' are restricted in Metadata field")
+			return
+		}
+	}
+	peerID := mux.Vars(r)["peerid"]
+	if uuid.Parse(peerID) == nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "Invalid peerID passed in url")
+		return
+	}
+
+	txn := transaction.NewTxn(ctx)
+	defer txn.Cleanup()
+	lock, unlock, err := transaction.CreateLockSteps(string(peerID))
+	if err != nil {
+		logger.WithError(err).WithField("peerid", peerID).Error("Failed to get lock on peer")
+		restutils.SendHTTPError(ctx, w, http.StatusNotFound, "Failed to get lock on peer")
+		return
+	}
+
+	txn.Steps = []*transaction.Step{
+		lock,
+		{
+			DoFunc: "peer-edit",
+			Nodes:  []uuid.UUID{gdctx.MyUUID},
+		},
+		unlock,
+	}
+	err = txn.Ctx.Set("peerid", peerID)
+	if err != nil {
+		logger.WithError(err).WithField("peerid", peerID).Error("Failed to set peerID in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error())
+		return
+	}
+	err = txn.Ctx.Set("req", req)
+	if err != nil {
+		logger.WithError(err).WithField("req-key", req).Error("Failed to set unmarshalled request information in transaction context")
```

Comment:
The variable `req` is a structure and not a string!

Change it as follows:
```go
// for set
logger.WithError(err).WithField("key", "req").Error("Failed to set key in transaction context")

// for get
logger.WithError(err).WithField("key", "req").Error("Failed to get key from transaction context")
```

Follow the same **everywhere else** wherever you call `txn.Set` or `txn.Get`. You don't have to specify the key in the error message. It's easier for it to be field. Error message won't change.

Reply:
@prashanthpai Is this for req only or should I do what you mentioned for peerid aswell

Reply:
Everywhere including `peer-id`

Reply:
@prashanthpai I am changing the error message, but  I am keeping the key for peer-id as "peer-id" as we discussed in office,